cmd/rig/cmd/capsule/instance/setup.go

```diff
@@ -82,6 +88,23 @@ func (c Cmd) Setup(parent *cobra.Command) {
 	logs.RegisterFlagCompletionFunc("follow", common.BoolCompletions)
 	instance.AddCommand(logs)
 
+	shell := &cobra.Command{
+		Use:               "shell [instance-id]",
+		Short:             "Open a shell to the instance",
+		Args:              cobra.MaximumNArgs(1),
+		RunE:              c.shell,
+		ValidArgsFunction: common.Complete(c.completions, common.MaxArgsCompletionFilter(1)),
+	}
+	shell.Flags().StringVar(&command, "command", "", "command to run in the shell")
```

Comment:
let's use `--` :
```
rig capsule shell my-instance -- sleep 10
```

that works automatically, if you use `args []string` for command line:

```
command := "sh"
arguments := append([]string{"-c". strings.Join(args, " "))
```

Potentially with a bit of quoting as well :)